tasks/basecalling/task_dorado_basecall.wdl

```diff
@@ -0,0 +1,80 @@
+version 1.0
+
+task basecall {
+  input {
+    File input_file                    # Single POD5 file for scatter processing
+    String dorado_model = "sup"         # Default model to 'sup', can be overridden with full model name see docs
+    String kit_name                    # Sequencing kit name
+    Int disk_size = 100
+    Int memory = 32
+    Int cpu = 8
+    String docker = "us-docker.pkg.dev/general-theiagen/staphb/dorado:0.8.0"
+  }
+
+  command <<< 
+    set -euo pipefail
+
+    # Capture Dorado version and log it
+    dorado --version > DORADO_VERSION 2>&1
+    echo "Captured Dorado version:" $(cat DORADO_VERSION)
+
+    # Define the model to use, substituting "sup" with the full model name if given
+    resolved_model="~{dorado_model}"
+    if [ "$resolved_model" = "sup" ]; then
+      resolved_model="dna_r10.4.1_e8.2_400bps_sup@v5.0.0"
+    fi
+
+    # Log the resolved model name
+    echo "Using Dorado model: $resolved_model"
+    echo "$resolved_model" > "DORADO_MODEL"
+
+    # Define a log file path to capture output
+    log_file="dorado_basecall.log"
+
+    # Create a unique output directory for each scatter job
+    base_name=$(basename "~{input_file}" .pod5)
+    sam_output="output/sam_${base_name}/"
+    mkdir -p "$sam_output"
+
+    echo "### Starting basecalling for ~{input_file} ###" | tee -a "$log_file"
+
+    # Set SAM file path with unique naming based on POD5 basename
+    sam_file="$sam_output/${base_name}.sam"
+
+    echo "Processing ~{input_file}, expected output: $sam_file" | tee -a "$log_file"
+
+    # Run Dorado basecaller and log output
+    dorado basecaller \
+      "~{dorado_model}" \
```

Comment:
I tried to override the `dorado_model` input parameter with the explicit fast model name `dna_r10.4.1_e8.2_400bps_fast@v5.0.0` and I got an error saying that the model file was not found:
```
[2024-11-16 15:47:11.731] [info] Running: "basecaller" "dna_r10.4.1_e8.2_400bps_fast@v5.0.0" "/mnt/miniwdl_task_container/work/_miniwdl_inputs/0/FAW95599_9a7a72fa_fad33a0d_0.pod5" "--kit-name" "SQK-RBK114-96" "--emit-sam" "--no-trim" "--output-dir" "output/sam_FAW95599_9a7a72fa_fad33a0d_0/" "--verbose"
[2024-11-16 15:47:11.743] [error] Model path does not exist at: 'dna_r10.4.1_e8.2_400bps_fast@v5.0.0' - Please download the model or use a model complex
[2024-11-16 15:47:11.743] [error] finalise() not called on a HtsFile.
```

since all the dorado model files are located at `/dorado_models` within the docker image filesystem this may need to be accounted for in the code. When I prefixed it and provided `/dorado_models/dna_r10.4.1_e8.2_400bps_fast@v5.0.0` it ran successfully.

So I'm thinking we need something like this in the code:
```
if [ "~{dorado_model}" == "fast" || "~{dorado_model}" == "hac" || "~{dorado_model}" == "sup" ]; then
  dorado_model=~{dorado_model}
  echo "DEBUG: dorado_model is set to: $dorado_model"
else
  dorado_model="/dorado_models/~{dorado_model}"
  echo "DEBUG: dorado_model is set to: $dorado_model"
fi
```
and then in the dorado command, need to swap `dorado basecaller ~{dorado_model}` with the bash variable `dorado basecaller $dorado_model` so that it handles both cases